core/src/test/kotlin/ch/ergon/dope/buildTest/QueryBuilderTest.kt

```diff
@@ -896,4 +901,30 @@ class QueryBuilderTest : ManagerDependentTest {
 
         assertEquals(expected, actual)
     }
+
+    @Test
+    fun `should support selecting let clause`() {
```

Comment:
I think we should define the let-clause in the following way:

```
val var1 = DopeVariable("var1", 5)
val var2 = DopeVariable("var2", "lel")
val variables = listOf(var1, var2)

create.select(...)
.from(...)
.withVariables(variables)
.where(var1.isEqualTo(7) //Typecheck
.and(var2.isEqualTo(7)) //Compiler error
```

Reply:
Better would be if the user doesn't need to pass the variables in the _.withVariables_ method but Dope will collect all variables used in the sub-clauses and add it in the let-part

Reply:
done